akka-actor-typed/src/main/scala/akka/actor/typed/internal/BehaviorImpl.scala

```diff
@@ -59,8 +59,11 @@ import scala.reflect.ClassTag
     onSignal:      PartialFunction[(SAC[T], Signal), Behavior[T]] = Behavior.unhandledSignal.asInstanceOf[PartialFunction[(SAC[T], Signal), Behavior[T]]])
     extends ExtensibleBehavior[T] {
 
-    override def receiveSignal(ctx: AC[T], msg: Signal): Behavior[T] =
-      onSignal.applyOrElse((ctx.asScala, msg), Behavior.unhandledSignal.asInstanceOf[PartialFunction[(SAC[T], Signal), Behavior[T]]])
+    override def receiveSignal(ctx: AC[T], msg: Signal): Behavior[T] = msg match {
+      // FIXME: ugh, now we have to make sure that every implementation of ExtensibleBehavior handles this signal. Is that realistic?
+      case DeferredMessage(cons) ⇒ receiveMessage(ctx, cons().asInstanceOf[T])
```

Comment:
I agree that the price and risk of forgetting to handle the DeferredMessage is too high and I see no way around it, ok we better do the safe thing then. I suggest that we catch-log-stop in the ActorAdapter to not even rely on that the parent will stop (will restart if typed actor is started by untyped parent?).

Reply:
Ok, makes sense.